mbuild/formats/hoomd3_forcefield.py

```diff
@@ -0,0 +1,402 @@
+import warnings
+import itertools
+import numpy as np
+import operator
+import packaging.version
+from collections import namedtuple
+
+import parmed as pmd
+import mbuild as mb
+from mbuild.utils.sorting import natural_sort
+from mbuild.utils.io import import_
+from mbuild.utils.conversion import RB_to_OPLS
+
+from .hoomd_snapshot import to_hoomdsnapshot
+
+hoomd = import_("hoomd")
+
+
+def create_hoomd3_forcefield(
+    structure,
+    ref_distance=1.0,
+    ref_mass=1.0,
+    ref_energy=1.0,
+    r_cut=1.2,
+    auto_scale=False,
+    snapshot_kwargs={},
+    pppm_kwargs={"Nx": 8, "Ny": 8, "Nz": 8, "order": 4},
+    init_snap=None
+):
+    """Convert a parametrized pmd.Structure to a HOOMD snapshot and forces.
+
+    Parameters
+    ----------
+    structure : parmed.Structure
+        ParmEd Structure object
+    ref_distance : float, optional, default=1.0
+        Reference distance for conversion to reduced units
+    ref_mass : float, optional, default=1.0
+        Reference mass for conversion to reduced units
+    ref_energy : float, optional, default=1.0
+        Reference energy for conversion to reduced units
+    r_cut : float, optional, default 1.2
+        Cutoff radius, in reduced units
+    auto_scale : bool, optional, default=False
+        Automatically use largest sigma value as ref_distance,
+        largest mass value as ref_mass,
+        and largest epsilon value as ref_energy
+    snapshot_kwargs : dict
+        Kwargs to pass to to_hoomdsnapshot
+    pppm_kwargs : dict
+        Kwargs to pass to hoomd's pppm function
+    init_snap : hoomd.data.SnapshotParticleData, optional, default=None
+        Initial snapshot to which to add the ParmEd structure object
+        (useful for rigid bodies)
+
+
+    Returns
+    ------
+    hoomd_snapshot : snapshot
+        HOOMD snapshot object to initialize the simulation
+    hoomd_forcefield : list
+        List of hoomd force computes created during conversion
+    ReferenceValues : namedtuple
+        Values used in scaling
```

Comment:
It would be helpful to list what type of objects are contained within `hoomd_forcefield`

and refer to the `hoomd_snapshot : ` type as more than just `snapshot`. I realize the description does offer more information, but something like `hoomd.snapshot` would be nice.